libsolidity/analysis/TypeChecker.cpp

```diff
@@ -359,6 +359,16 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 	};
 	for (ASTPointer<VariableDeclaration> const& var: _function.parameters())
 	{
+		TypePointer baseType = type(*var);
+		while (auto const* arrayType = dynamic_cast<ArrayType const*>(baseType.get()))
+			baseType = arrayType->baseType();
```

Comment:
Don't we have a helper on the type for this?

Reply:
I don't see one. In ``Type::fullEncodingType`` the same is done, but also inlined - it's only two lines, not sure whether a helper for that makes sense. I could add one, but what to call it and where to put it? A member in ``ArrayType`` wouldn't be too helpful, because then I still have to cast and check once, which would be almost as long as the whole thing - and a (probably then even virtual) member of ``Type`` seems like overkill...